WMF Framework/EventLoggingService.swift

```diff
@@ -286,38 +288,34 @@ public class EventLoggingService : NSObject, URLSessionDelegate {
     
     @objc
     private func tryPostEvents() {
+        let operation = AsyncBlockOperation { (operation) in
+            let moc = self.managedObjectContext
+            moc.perform {
+                let fetch: NSFetchRequest<EventRecord> = EventRecord.fetchRequest()
+                fetch.sortDescriptors = [NSSortDescriptor(keyPath: \EventRecord.recorded, ascending: true)]
+                fetch.predicate = NSPredicate(format: "(posted == nil) AND (failed != TRUE)")
+                fetch.fetchLimit = self.postBatchSize
+
 
-        self.postLock.lock()
-        guard started, !posting else {
-            self.postLock.unlock()
-            return
-        }
-        posting = true
-        self.postLock.unlock()
-        
-        let moc = self.managedObjectContext
-        moc.perform {
-            let fetch: NSFetchRequest<EventRecord> = EventRecord.fetchRequest()
-            fetch.sortDescriptors = [NSSortDescriptor(keyPath: \EventRecord.recorded, ascending: true)]
-            fetch.predicate = NSPredicate(format: "(posted == nil) AND (failed != TRUE)")
-            fetch.fetchLimit = self.postBatchSize
-            
-            do {
                 var eventRecords: [EventRecord] = []
-                defer {
-                    if eventRecords.count > 0 {
-                        self.postEvents(eventRecords)
-                    } else {
-                        self.postLock.lock()
-                        self.posting = false
-                        self.postLock.unlock()
+                do {
+                    defer {
```

Comment:
If code is added after this do/catch block, it'd be called after this `defer` statement. I didn't know this was the case, but tested it with a playground:

```
do {
    defer {
        print("defer")
    }
}

sleep(5)
print("ok")
```

Since there's no early return to necessitate the defer, it'd be clearer to add the contents of this `defer` block after the do/catch